src/dlgTriggerEditor.cpp

```diff
@@ -4738,56 +4799,168 @@ void dlgTriggerEditor::slot_var_selected(QTreeWidgetItem *pItem)
 
     mpCurrentActionItem = pItem; //remember what has been clicked to save it
     int ID = pItem->data(0,Qt::UserRole).toInt();
     TAction * pT = mpHost->getActionUnit()->getAction(ID);
     if( pT )
     {
+        // We set every 'field' up even though they may not be appropriate for
+        // this particular item...
         mpActionsMainArea->lineEdit_action_name->setText( pT->getName() );
         mpActionsMainArea->checkBox_pushdownbutton->setChecked( pT->isPushDownButton() );
-        mpActionsMainArea->lineEdit_action_icon->setText( pT->getIcon() );
+        QString iconFileName = pT->getIcon();
+        if( iconFileName.isEmpty() )
+        {
+            mpActionsMainArea->lineEdit_action_icon->clear();
+            mpActionsMainArea->pushButton_changeIcon->setIcon( QIcon() );
+            mpActionsMainArea->pushButton_clearIcon->setEnabled( false );
+        }
+        else
+        {
+            mpActionsMainArea->lineEdit_action_icon->setText( iconFileName );
+            mpActionsMainArea->pushButton_changeIcon->setIcon( QIcon( iconFileName ) );
+            mpActionsMainArea->pushButton_clearIcon->setEnabled( true );
+        }
+        mpActionsMainArea->lineEdit_action_command_down->setText( pT->getCommandButtonDown() );
+        mpActionsMainArea->lineEdit_action_command_up->setText( pT->getCommandButtonUp() );
         mpSourceEditor->setPlainText( pT->getScript() );
         // location = 1 = location = bottom is no longer supported
         int location = pT->mLocation;
-        if( location > 0 ) location--;
+        if( location > 0 )
+        {
+            location--;
+        }
         mpActionsMainArea->comboBox_location->setCurrentIndex( location );
         mpActionsMainArea->comboBox_orientation->setCurrentIndex( pT->mOrientation );
-        QColor color = pT->getButtonColor();
-        QPalette palette;
-        palette.setColor( QPalette::Button, color );
-        mpActionsMainArea->buttonRotation->setCurrentIndex( pT->getButtonRotation() );
-        mpActionsMainArea->buttonColumns->setValue( pT->getButtonColumns() );
-        mpActionsMainArea->css->clear();
+        mpActionsMainArea->comboBox_buttonRotation->setCurrentIndex( pT->getButtonRotation() );
+        mpActionsMainArea->spinBox_buttonColumns->setValue( pT->getButtonColumns() );
         mpActionsMainArea->css->setPlainText( pT->css );
-        if( ! pT->getParent() )
+
+        // Now show/hide the relevant bits:
+        mpActionsMainArea->show();
+        mpSystemMessageArea->hide();
+        if( pT->isFolder() && ! pT->mPackageName.isEmpty() )
         {
+            // This is a module master or package folder - hide everything but
+            // the name and protect that from editing
+            mpActionsMainArea->widget_middle->hide();
+            mpActionsMainArea->widget_bottom->hide();
+            mpSourceEditorArea->hide();
+            mpActionsMainArea->lineEdit_action_name->setReadOnly(true);
```

Comment:
Trouble is, disabling the widget deactivates the tooltip functionality as well (on some, if not all platforms, I'm not sure), which is a bitch if you want to explain why it is disabled ... :cry: